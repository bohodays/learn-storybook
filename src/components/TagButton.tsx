import React from "react";

interface ITagButtonProps {
  isChecked: boolean;
  children: string;
  onClick: () => void;
}

const TagButton = ({ isChecked, children, onClick }: ITagButtonProps) => {
  const buttonStyle = isChecked
    ? "bg-white text-primary"
    : "bg-dark-opacity text-white";

  return (
    <button
      className={`h-[33px] rounded-tag-button px-[10px] border border-white text-sm font-medium ${buttonStyle}`}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

export default TagButton;
