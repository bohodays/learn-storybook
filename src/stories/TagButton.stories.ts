import type { Meta, StoryObj } from "@storybook/react-vite";

import { fn } from "storybook/test";
import TagButton from "../components/TagButton";

// Docs 정보
const meta = {
  title: "Buttons/TagButton", // 경로를 나타냄
  component: TagButton,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: "text",
      desciption: "버튼의 text",
      defaultValue: "Button",
    },
    isChecked: {
      control: "boolean",
      description: "버튼 활성화 여부",
      defaultValue: false,
    },
    onClick: { action: "clicked", description: "버튼 클릭 이벤트" },
  },
} satisfies Meta<typeof TagButton>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    isChecked: false,
    children: "Button",
    onClick: () => {},
  },
};
