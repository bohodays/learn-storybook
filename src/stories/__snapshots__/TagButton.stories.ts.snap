// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons/TagButton Default smoke-test 1`] = `
<button class="h-[33px] rounded-tag-button px-[10px] border border-white text-sm font-medium bg-dark-opacity text-white">
  Button
</button>
`;
